posts/openscanhub-prototype/index.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "SAST using OpenScanHub is here!"
+date: 2024-07-30T08:46:18+02:00
+authors: lbarczio
+tags:
+  - copr
+  - srpm
+  - sast
+  - openscanhub
+  - shift left
+---
+
+We are excited to announce a new experimental feature in our service: the integration of Static
+Application Security Testing (SAST) using [OpenScanHub](https://openscanhub.fedoraproject.org/).
+
+OpenScanHub is a service that runs various static analyzers on RPM packages - by default `Cppcheck`,
+`ShellCheck`, the static analyzers embedded in `GCC` and `Clang`, and the `find-unicode-control` tool.
```

Comment:
As far as I know, @siteshwar has disabled Clang Analyzer in the default scanning profile because it was missing fundamental features, such as the support for `__attribute__((cleanup))`, which resulted in negative feedback from Fedora developers: https://github.com/llvm/llvm-project/issues/4260

Reply:
It is disabled as of now, but is enabled by default and can be enabled any time in the future. Also, I am yet to enable `find-unicode-control`, but that should not stop us from listing them as potential analyzers.

Reply:
> It is disabled as of now, but is enabled by default

I do not understand the above sentence.  Either the default scanning profile contains the tool or not.  I checked a recent OSH task triggered by Packit and it did not use Clang Analyzer according to scan properties: https://openscanhub.fedoraproject.org/task/2459/log/containers-common-0.60.0-1.20240731120946597433.pr2107.7.g12f3adf3/scan.ini

Reply:
If you enable it in the future, it can be advertised in a future blog post.  But saying in this blog post that `clang` is used by default, despite you disabled it a few weeks ago, is misleading.

Reply:
agreed, will change it. I took it from https://openscanhub.fedoraproject.org/ so it should be probably changed there as well.

Reply:
In that case, please remove `find-unicode-control` too.

Reply:
On a second thought, can we change "by default uses"  to "by default can use" and list all the analyzers OSH can use. While this sentence is technically correct right now, it is subject to change. And we are showing a relatively limited capability of OSH. We can probably mention it in the next sentence about which analyzers are enabled right now.